Userland/Libraries/LibJS/Runtime/ScriptFunction.cpp

```diff
@@ -57,8 +67,16 @@ void ScriptFunction::initialize(GlobalObject& global_object)
     auto& vm = this->vm();
     Function::initialize(global_object);
     if (!m_is_arrow_function) {
-        Object* prototype = vm.heap().allocate<Object>(global_object, *global_object.new_script_function_prototype_object_shape());
-        prototype->define_property(vm.names.constructor, this, Attribute::Writable | Attribute::Configurable);
+        auto* prototype = vm.heap().allocate<Object>(global_object, *global_object.new_script_function_prototype_object_shape());
```

Comment:
This doesn't seem correct for `FunctionKind::Generator` - this way `(function () {}).prototype === (function* () {}).prototype`, right?

Reply:
I agree in that it doesn't seem correct, but it actually is. In the case of a normal function, the `prototype` property is just a plain object that points back to the function with its `constructor` property, and whose internal prototype is the object prototype (I think). In the case of a generator function, however, it's `prototype` property points to this weird intermediate object which in the actual prototype of the iterator returned from the generator function when you call it. That object's internal prototype is `%Generator.prototype.prototype%`.

It's all very, very confusing, and I probably didn't do that great of a job explaining it. You've really got to stare at the figure I linked to understand everything.

Reply:
> I probably didn't do that great of a job explaining it

You did a better job than the figure tbh! I did look at it, but I misunderstood the `prototype -> g1.prototype ->` part - all good then :)